webapp/src/containers/HomePageContainer.js

```diff
@@ -13,25 +13,30 @@ class HomePageContainer extends React.Component {
   static propTypes = {
     connectWeb3: PropTypes.func,
     parcelStates: stateData(PropTypes.object).isRequired,
-    addressState: stateData(PropTypes.object).isRequired
+    addressState: stateData(PropTypes.object).isRequired,
+    shiftNeverPressed: PropTypes.bool.isRequired
   }
 
   componentWillMount() {
     this.props.connectWeb3()
   }
 
   render() {
-    const { parcelStates, addressState } = this.props
+    const { shiftNeverPressed, parcelStates, addressState } = this.props
     const requiredDataReady = !isEmptyObject(parcelStates) && addressState.data
 
-    return <HomePage requiredDataReady={!!requiredDataReady} />
+    return <HomePage
+      requiredDataReady={!!requiredDataReady}
+      shiftNeverPressed={shiftNeverPressed}
```

Comment:
I might be overthinking this but this looks like the wrong component for this behaviour, same goes for the window event on `store.js`. Why not having a `<ShiftNotificationContainer />` which does something like:

```javascript
const SHIFT = 'Shift'

class ShiftNotificationContainer extends React.Component {
  static propTypes = {
    shiftNeverPressed: PropTypes.bool.isRequired
  }

  componentWillMount() {
   window.addEventListener('keydown', (event) => {
     if (event.key === SHIFT) {
       store.dispatch({ type: types.shift.down })
     }
   })
   
   window.addEventListener('keyup', (event) => {
     if (event.key === SHIFT) {
       store.dispatch({ type: types.shift.up })
     }
   })
  }

  componentWillUnmount() {
    // remove listeners from window
  }

  render() {
    const { shiftNeverPressed } = this.props

    return {shiftNeverPressed && <ShiftNotification />
  }
}
```

What do you think?